api/s2n.h

```diff
@@ -1076,24 +1076,120 @@ struct s2n_async_pkey_op;
 typedef enum { S2N_ASYNC_PKEY_VALIDATION_FAST, S2N_ASYNC_PKEY_VALIDATION_STRICT } s2n_async_pkey_validation_mode;
 typedef enum { S2N_ASYNC_DECRYPT, S2N_ASYNC_SIGN } s2n_async_pkey_op_type;
 
+/**
+ * Callback function for handling private key operations
+ *
+ * Invoked every time an operation requiring the private key is encountered
+ * during the handshake.
```

Comment:
I don't think we've added support for client-side and mutual auth, right?

Reply:
We actually added that support as part of the private key offloading work. You should be able to run s2n-tls completely without a private key now, if you're using private key offloading.

Reply:
Oh, nice! 👍